### PR TITLE
Validator for move params

### PR DESCRIPTION
**Pull request by ngw**
Fixes https://dsdmoj.atlassian.net/browse/P4-701

### CODE REVIEW

**Comment by ngw** [comment]
Force pushed indentation and spacing to make rubocop happy

File: app/controllers/api/v1/moves_controller.rb [hunk 1]
```diff
@@ -4,9 +4,14 @@ module Api
...
-        import_moves_from_nomis
-        moves = Moves::Finder.new(filter_params).call
-        paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES
+        moves_params = Moves::ParamsValidator.new(params[:filter])
+        if moves_params.valid?
+          import_moves_from_nomis
+          moves = Moves::Finder.new(filter_params).call
+          paginate moves, include: MoveSerializer::INCLUDED_ATTRIBUTES
+        else
+          render json: { error: moves_params.errors }
```
Comment (stevehook) [inline app/controllers/api/v1/moves_controller.rb]: I think there should be a non-200 status code in here somewhere

Comment (stevehook) [reply app/controllers/api/v1/moves_controller.rb]: Also can you adapt this to fit with the JSONAPI spec - https://jsonapi.org/format/#errors ?

I think there are a couple of examples elsewhere in the code. Maybe you can reuse https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/controllers/api_controller.rb#L40 ?

Comment (stevehook) [reply app/controllers/api/v1/moves_controller.rb]: You might need to change the Swagger docs depending on whether this case is covered by an existing scenario.

**Pull request merged**